mr/src/main/java/org/apache/iceberg/mr/IcebergInputFormat.java

```diff
@@ -0,0 +1,567 @@
+      if (reuseContainers) {
+        parquetReadBuilder.reuseContainers();
+      }
+
+      switch (inMemoryDataModel) {
+        case PIG:
+        case HIVE:
+          //TODO implement value readers for Pig and Hive
+          throw new UnsupportedOperationException();
+        case DEFAULT:
+          parquetReadBuilder.createReaderFunc(
+              fileSchema -> GenericParquetReaders.buildReader(readSchema, fileSchema));
+      }
+      return parquetReadBuilder.build();
+    }
+
+    private CloseableIterable<T> newOrcIterable(InputFile inputFile, FileScanTask task, Schema readSchema) {
+      ORC.ReadBuilder orcReadBuilder = ORC.read(inputFile)
+                                          .schema(readSchema)
+                                          .caseSensitive(caseSensitive)
+                                          .split(task.start(), task.length());
+      // ORC does not support reuse containers yet
+      switch (inMemoryDataModel) {
+        case PIG:
+        case HIVE:
+          //TODO implement value readers for Pig and Hive
+          throw new UnsupportedOperationException();
+        case DEFAULT:
+          //TODO: We do not have support for Iceberg generics for ORC
+          throw new UnsupportedOperationException();
+      }
+
+      return orcReadBuilder.build();
+    }
+  }
+
+  private static Table findTable(Configuration conf) {
+    String path = conf.get(TABLE_PATH);
+    String catalogFuncClass = conf.get(CATALOG);
+    if (catalogFuncClass != null) {
+      Function<Configuration, Catalog> catalogFunc
+          = (Function<Configuration, Catalog>)
+          DynConstructors.builder(Function.class)
+                         .impl(catalogFuncClass)
+                         .build()
+                         .newInstance();
+      Catalog catalog = catalogFunc.apply(conf);
+      TableIdentifier tableIdentifier = TableIdentifier.parse(path);
+      return catalog.loadTable(tableIdentifier);
+    } else if (path.contains("/")) {
+      HadoopTables tables = new HadoopTables(conf);
+      return tables.load(path);
+    } else {
+      throw new IllegalArgumentException("No custom catalog specified to load table " + path);
+    }
+  }
+
+  private static class IcebergSplit extends InputSplit implements Writable {
+    private static final String[] ANYWHERE = new String[]{"*"};
+    private CombinedScanTask task;
+    private transient String[] locations;
+    private transient Configuration conf;
+
+    IcebergSplit(Configuration conf, CombinedScanTask task) {
+      this.task = task;
+      this.conf = conf;
+    }
+
+    @Override
+    public long getLength() {
+      return task.files().stream().mapToLong(FileScanTask::length).sum();
+    }
+
+    @Override
+    public String[] getLocations() {
+      boolean localityPreferred = conf.getBoolean(LOCALITY, false);
+      if (!localityPreferred) {
+        return ANYWHERE;
```

Comment:
Does MR treat "*" as `ANYWHERE`?

Reply:
I copied this from IcebergPigInputFormat. I'll double check